tools/protodoc/protodoc.py

```diff
@@ -691,6 +699,12 @@ def visit_file(self, file_proto, type_context, services, msgs, enums):
         if all(len(msg) == 0 for msg in msgs) and all(len(enum) == 0 for enum in enums):
             has_messages = False
 
+        v2_link = ""
+        if annotations.REDIRECT_ANNOTATION in type_context.source_code_info.file_level_annotations:
+            v2_link = f"\n\n.. sidebar:: API v3\n\n    :ref:`Legacy api v2 documentation <v{ENVOY_LAST_V2_VERSION}:envoy_api_file_{type_context.source_code_info.file_level_annotations[annotations.REDIRECT_ANNOTATION]}>`\n\n"
```

Comment:
Worth pointing out that the v2 API is not just legacy but unimplemented.

Reply:
this has been updated, a selection of links that include the updated header:

- https://storage.googleapis.com/envoy-pr/230e00d/docs/api-v3/config/bootstrap/v3/bootstrap.proto.html
- https://storage.googleapis.com/envoy-pr/230e00d/docs/api-v3/config/listener/v3/listener_components.proto.html
- https://storage.googleapis.com/envoy-pr/230e00d/docs/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto.html

also - see #16329 for a follow-up to try and improve the rendering here